update-dependencies/Program.cs

```diff
@@ -3,152 +3,138 @@
-            {
-                if (string.Equals(arg, "--Update", StringComparison.OrdinalIgnoreCase))
-                {
-                    s_updateOnly = true;
-                }
-                else if (arg.Contains('='))
-                {
-                    int delimiterIndex = arg.IndexOf('=');
-                    string name = arg.Substring(0, delimiterIndex);
-                    string value = arg.Substring(delimiterIndex + 1);
-
-                    Environment.SetEnvironmentVariable(name, value);
-                }
-                else
-                {
-                    Console.Error.WriteLine($"Unrecognized argument '{arg}'");
-                    return false;
-                }
-            }
-
-            return true;
+            Environment.Exit(0);
         }
 
-        private static DependencyUpdateResults UpdateFiles()
+        private static async Task<DependencyUpdateResults> UpdateFiles()
         {
-            // Ideally this logic would depend on the CLI produces and consumes metadata.  Since it doesn't
-            // exist various version information is inspected to obtain the latest CLI version along with
-            // the runtime (e.g. shared framework) it depends on.
-
-            BuildInfo cliBuildInfo = BuildInfo.Get(CliBuildInfoName, s_config.CliVersionUrl, fetchLatestReleaseFile: false);
-            // Adjust the LatestReleaseVersion since it is not the full version and all consumers here need it to be.
-            cliBuildInfo.LatestReleaseVersion = $"{s_config.RuntimeReleasePrefix}-{cliBuildInfo.LatestReleaseVersion}";
-            CliDependencyHelper cliDependencyHelper = new CliDependencyHelper(cliBuildInfo.LatestReleaseVersion);
-            string sharedFrameworkVersion = cliDependencyHelper.GetSharedFrameworkVersion();
-
-            IEnumerable<DependencyBuildInfo> buildInfos = new[]
+            Trace.TraceInformation($"Retrieving build info from '{Options.BuildInfoUrl}'");
+            Stream stream = await (new HttpClient().GetStreamAsync(Options.BuildInfoUrl));
+            XDocument buildInfoXml = XDocument.Load(stream);
+            OrchestratedBuildModel buildInfo = OrchestratedBuildModel.Parse(buildInfoXml.Root);
+            BuildIdentity sdkBuild = buildInfo.Builds
+                .First(build => string.Equals(build.Name, "cli", StringComparison.OrdinalIgnoreCase));
+            BuildIdentity coreSetupBuild = buildInfo.Builds
+                .First(build => string.Equals(build.Name, "core-setup", StringComparison.OrdinalIgnoreCase));
+
+            // TODO:  Remove once ProductVersion is written to build-info
+            EndpointModel endpointInfo = buildInfo.Endpoints
+                .First(endpoint => string.Equals(endpoint.Id, "dotnetcli", StringComparison.OrdinalIgnoreCase));
+            BlobArtifactModel blobInfo = endpointInfo.Artifacts.Blobs
+                .First(artifact => artifact.Id.StartsWith("Runtime", StringComparison.OrdinalIgnoreCase));
```

Comment:
Just want to point out that using this endpoint means you want to trigger on `installers.semaphore`, since that indicates the endpoint was published to and added to the manifest.

Reply:
Yep - I that is was I was thinking needed to happen - thanks for confirming.